modules/database/database.bal

```diff
@@ -546,13 +554,20 @@ public isolated transactional function updateExperiment(int experimentId, *Exper
 # Get the number of data entries for a specific experiment.
 #
 # + experimentId - The experiment id
+# + search - search keyword in name, data type and content type (insensitive)
 # + all - If true count all experiment data including old version, if false count only the newest verwions (e.g. distinct data names)
 # + return - The count or the encountered error
-public isolated transactional function getExperimentDataCount(int experimentId, boolean all = true) returns int|error {
+public isolated transactional function getExperimentDataCount(int experimentId, string? search, boolean all = true) returns int|error {
     stream<RowCount, sql:Error?> result;
     if all {
+        if search != () && search != "" {
+            result = experimentDB->query(`SELECT count(*) AS rowCount FROM ExperimentData WHERE experimentId = ${experimentId} AND ((name LIKE '%${search}%') OR (type  LIKE '%${search}%') OR (contentType LIKE '%${search}%'));`);
```

Comment:
Refactor out the filter creation into their own functions and reuse them (possibly into their own files) and use concat query to add the filter to the query. Also this filter still uses the failing pattern for like queries.